public/js/add-post.js

```diff
@@ -0,0 +1,38 @@
+const createPost = async (postTitle, postContent) => {
+  let res = await fetch("/api/posts", {
+    method: "POST",
+    body: JSON.stringify({
+      title: postTitle,
+      content: postContent,
+    }),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  if (res.status === 200) {
+    // Post created successfully so now go to the dashboard
+    document.location.replace("/dashboard");
+    //const data = await res.json();
+    //console.log(data);
+  } else {
+    alert("Something went wrong while creating a post on server sideΩ");
+  }
+};
+
+let createPostBtn = document.getElementById("create-post");
+createPostBtn.addEventListener("click", (e) => {
+  let postTitle = document.getElementById("post-title").value.trim();
```

Comment:
consider chaining the event listener onto line 22 and placing this logic in a separate function for better readability.